lnwallet/channel.go

```diff
@@ -6028,3 +6038,107 @@ func (lc *LightningChannel) RemoteCommitHeight() uint64 {
 func (lc *LightningChannel) FwdMinHtlc() lnwire.MilliSatoshi {
 	return lc.channelState.LocalChanCfg.MinHTLC
 }
+
+// calcIncomingHtlcs calculates the number of committed and uncommitted HTLC's
+// from the PoV of the tip of the commitment chain. This function is used to
+// determine if one of the counterparty's adds will overflow the commitment
+// transaction before it happens in ReceiveNewCommitment. Since these Adds will
+// be appended to the tip of the chain, the tail is not used. It takes an
+// htlcID parameter that is used to determine the number of uncommitted HTLC's
+// in-memory.
+func (lc *LightningChannel) calcIncomingHtlcs(htlcID uint64) uint64 {
+	// There are two cases to consider when calculating the true number of
```

Comment:
This is starting to become quite complex. Could we instead call `validateCommitmentSanity`, which I believe already performs these checks?

Reply:
Yup, cuts down the whole diff pretty much